cpp/src/reshape/byte_cast.cu

```diff
@@ -80,13 +81,20 @@ struct byte_list_conversion {
 
     rmm::device_buffer null_mask = detail::copy_bitmask(input_column, stream, mr);
 
-    return make_lists_column(input_column.size(),
-                             std::move(offsets_column),
-                             std::move(byte_column),
-                             input_column.null_count(),
-                             std::move(null_mask),
-                             stream,
-                             mr);
+    auto result = make_lists_column(input_column.size(),
+                                    std::move(offsets_column),
+                                    std::move(byte_column),
+                                    input_column.null_count(),
+                                    std::move(null_mask),
+                                    stream,
+                                    mr);
```

Comment:
I had this doubt for a while. Must all temporaries be created with default memory resource only or is it okay to create temporaries using `mr` too sometimes?

Reply:
This isn't a temporary. We need to use `mr` if the result can be returned to the user.

The `result` column _may_ be returned as-is, if no nulls need to be purged. If there are nulls to purge, then the purged column uses `mr` and is returned (leaving `result` to be deallocated when it goes out of scope).